src/codegen/estree/nodes.ts

```diff
@@ -3,8 +3,11 @@ import { TSESTree as ts } from '@typescript-eslint/types'
 // Since we're generating our own AST we don't have any positional information, but
 // the types require it. To fix it, we spread this dummy object into the nodes we create.
 export const baseProps = {
-  loc: null as unknown as ts.SourceLocation,
-  range: null as unknown as ts.Range,
+  loc: {
+    start: { line: 0, column: 0 },
+    end: { line: 0, column: 0 },
+  },
+  range: [0, 1] as [number, number],
```

Comment:
Prettier would go into an infinite loop when emitting arrays without some kind of location information. Luckily, it doesn't care that everything is seemingly contained in a single character in the doc.